projects/kit/components/input-slider/input-slider.component.ts

```diff
@@ -133,149 +178,133 @@ export class TuiInputSliderComponent
         );
     }
 
-    @HostBinding('class._has-tooltip')
-    get hasTooltip(): boolean {
-        return !!this.hintController.content && !this.disabled;
+    get computedSteps(): number {
+        return this.steps || (this.max - this.min) / this.quantum;
     }
 
-    @HostBinding('class._min-label')
-    get showMinLabel(): boolean {
-        return !this.focused && this.value === this.min && !!this.minLabel;
+    get precision(): number {
+        return getFractionPartPadded(this.quantum).length;
     }
 
-    @HostBinding('class._max-label')
-    get showMaxLabel(): boolean {
-        return !this.focused && this.value === this.max && !!this.maxLabel;
+    get decimal(): TuiDecimalT {
+        return this.precision ? 'not-zero' : 'never';
     }
 
-    get computedValue(): string {
-        if (this.focused && this.isInputValueNotFinished) {
-            return this.inputValue;
+    /**
+     * TODO remove old property `size` in v3.0
+     */
+    get computedSize(): TuiSizeL {
+        if (this.isNew) {
+            tuiAssert.assert(
+                this.controller.size !== 's',
+                "Size 's' is not supported by this input.",
+            );
+
+            return this.controller.size === 'l' ? 'l' : 'm';
         }
 
-        return this.formattedValue;
+        return this.size;
     }
 
-    get showValue(): boolean {
-        return !this.showMinLabel && !this.showMaxLabel;
+    /**
+     * @deprecated for backward compatibility
+     * TODO replace by just `this.valueContent` in v3.0
+     */
+    get computedValueContent(): PolymorpheusContent<TuiContextWithImplicit<number>> {
+        return this.minLabel || this.maxLabel
+            ? legacyMinMaxLabel(this)
+            : this.valueContent;
     }
 
-    get inputValue(): string {
-        return this.focusableElement ? this.focusableElement.nativeElement.value : '';
+    @tuiPure
+    computeContext(value: number): TuiContextWithImplicit<number> {
```

Comment:
This is unnecessary, just write inline object in template, Angular optimizes it.

Reply:
fixed